js/timeout-transition-group.jsx

```diff
@@ -87,6 +84,40 @@ function animationSupported() {
     return endEvents.length !== 0;
 }
 
+/**
+ * Functions for element class management to replace dependency on jQuery
+ * addClass, removeClass and hasClass
+ */
+function addClass(element, className) {
+    if (element.classList) {
+        element.classList.add(className);
+    } else if (!hasClass(element, className)) {
+        element.className = element.className + ' ' + className;
+    }
+    return element;
+}
+function removeClass(element, className) {
+    if (hasClass(className)) {
+        if (element.classList) {
+            element.classList.remove(className);
+        } else {
+            element.className = element.className
+                .replace(className, '')
+                .replace(new RegExp('/(^|\\s)' + className + '($|\\s)/'), ' ')
```

Comment:
From my understanding, special characters such as "[]$" etc can be in class names, you just need to escape them to write them in an actual css file :(. This could theoretically get us into trouble here if someone put one of those characters in their class name (although they really shouldn't).

To be thorough, could we either escape the className here (MDN has a [function for escaping regexes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions)), or use raw string methods, such as:

    (" " + element.className + " ").replace(" " + className + " ", " ").trim()

?